cmd/collectors/rest/plugins/qtree/qtree.go

```diff
@@ -338,12 +254,27 @@ func (q *Qtree) handlingQuotaMetrics(result []gjson.Result, quotaCount *int, num
 				q.Logger.Debug().Msgf("add (%s) instance: %v", attribute, err)
 				return err
 			}
+
+			// In 22.05, populate metrics with qtree labels
+			if q.historicalLabels {
+				for _, label := range q.data.GetExportOptions().GetChildS("instance_keys").GetAllChildContentS() {
+					if val := qtreeInstance.GetLabel(label); val != "" {
+						quotaInstance.SetLabel(label, val)
+					}
+				}
+			}
+
 			// set labels
 			quotaInstance.SetLabel("type", quotaType)
 			quotaInstance.SetLabel("qtree", tree)
 			quotaInstance.SetLabel("volume", volume)
 			quotaInstance.SetLabel("svm", vserver)
 
+			// If the Qtree is the volume itself, then qtree label is empty, so copy the volume name to qtree.
```

Comment:
Did you mean to bring this change over? This was not happening in the original handlingQuotaMetrics